src/db/duty-collection.js

```diff
@@ -0,0 +1,167 @@
+import { getCollection } from "./client.js";
+
+async function insertDuty(newDuty) {
+	await getCollection("duties").insertOne(newDuty);
+}
+
+async function getDuties(filter) {
+	const duties =
+		Object.keys(filter).length > 0 ? await getCollection("duties").find(filter).toArray() : [];
+
+	return duties;
+}
+
+async function getDutyByID(dutyID) {
+	const duty = await getCollection("duties").findOne({ _id: ObjectId.createFromHexString(dutyID) });
+
+	return duty;
+}
+
+async function deleteDutyByID(dutyID) {
+	await getCollection("duties").deleteOne({
+		_id: ObjectId.createFromHexString(dutyID),
+	});
+}
+
+async function patchDuty(dutyID, update) {
+	const updatedDuty = await getCollection("duties").findOneAndUpdate(
+		{ _id: ObjectId.createFromHexString(dutyID) },
+		{ $set: update, $currentDate: { updatedAt: true } },
+		{ returnDocument: "after" },
+	);
+
+	return updatedDuty;
+}
+
+async function putConstraints(dutyID, newConstraints) {
+	const updatedDuty = await getCollection("duties").findOneAndUpdate(
+		{ _id: ObjectId.createFromHexString(dutyID) },
+		{
+			$addToSet: { constraints: { $each: newConstraints } },
+			$currentDate: { updatedAt: true },
+		},
+		{ returnDocument: "after" },
+	);
+
+	return updatedDuty;
+}
+
+async function getOverlappingDutySoldiers(duty) {
+	const overlappingSoldiers = await getCollection("duties")
+		.aggregate([
+			{
+				$match: {
+					_id: { $ne: duty._id },
+					startTime: { $lt: duty.endTime },
+					endTime: { $gt: duty.startTime },
+				},
+			},
+			{ $unwind: "$soldiers" },
+			{ $project: { _id: 0, soldierId: "$soldiers" } },
+		])
+		.toArray();
+
+	return overlappingSoldiers;
+}
+
+async function getAvailableSoldiersForDuty(duty, soldierFilter) {
+	const availableSoldiers = await getCollection("soldiers")
+		.aggregate([
+			{ $match: soldierFilter },
+			{
+				$lookup: {
+					from: "duties",
+					let: { soldierId: "$_id" },
+					pipeline: [
+						{
+							$match: {
+								_id: { $ne: duty._id },
+								startTime: { $lt: duty.endTime },
+								endTime: { $gt: duty.startTime },
+							},
+						},
+						{
+							$match: {
+								$expr: {
+									$in: ["$$soldierId", "$soldiers"],
+								},
+							},
+						},
+					],
+					as: "conflictingDuties",
+				},
+			},
+			{
+				$match: {
+					conflictingDuties: { $size: 0 },
+				},
+			},
+		])
+		.toArray();
+
+	return availableSoldiers;
+}
+
+async function addSoldiersToDuty(dutyID, scheduledSoldiers) {
+	const updatedDuty = await getCollection("duties").findOneAndUpdate(
+		{ _id: ObjectId.createFromHexString(dutyID) },
+		{
+			$addToSet: { soldiers: { $each: scheduledSoldiers } },
+			$set: { status: "scheduled" },
+			$push: {
+				statusHistory: {
+					status: "scheduled",
+					date: new Date(),
+				},
+			},
+			$currentDate: { updatedAt: true },
+		},
+		{ returnDocument: "after" },
+	);
+
+	return updatedDuty;
+}
+
+async function cancelDuty(dutyID) {
```

Comment:
in this layer functions names should explain what you are doing with the documents at the database level, usually starting with `update`/`insert`/`delete`/`push`/`upsert`...

```suggestion
async function updateDutyToCanceled(dutyID) {
```